src/helpers.ts

```diff
@@ -525,14 +554,18 @@ export function getBootOrderDevices(vm) {
         }
     }
 
+    function to_num(str: opt_string): number | undefined {
+        return str ? Number(str) : undefined;
+    }
```

Comment:
brrrr, can't we default to something sane instead of "unknown"? If undefined like last in the list?

Reply:
There is no `unknown` here.  I think this is better than what we had... the old code was freely mixing strings and numbers and did arithmetic on strings. But bootOrder can also be "undefined". I think it is better to explicitly convert to numbers before doing arithmetic.

I have renamed this to `to_optNumber`, is that better?